go/arrow/array/compare.go

```diff
@@ -630,6 +631,34 @@ func validityBitmapEqual(left, right arrow.Array) bool {
 	return true
 }
 
+func stripNulls(s string) string {
+	return strings.ReplaceAll(s, "\x00", "")
+}
+
```

Comment:
In the Arrow spec a String array should be all valid utf8. So shouldn't this only be for trimming trailing Nulls? (ie: use `strings.TrimRight` instead of `strings.ReplaceAll` )?

To this end, can you have a test with a string that contains a null inside it that *isn't* at the end and ensure that this approxequal only trims nulls, and doesn't strip them from within the string?

Reply:
Interesting, wasn't aware of the valid utf8 specification. In that case there is no check in `string.Append` [method](https://github.com/cloudquery/arrow/blob/main/go/arrow/array/string.go#L321) to make sure there are no nulls inside the string. Should we add this check or string nulls inside the string? wdyt?

Reply:
So for performance reasons we don't validate the utf8 strings on Append currently and leave it up to the producer to ensure that they are passing valid utf-8 strings when constructing the array (if it's not valid utf-8 they should be using a Binary array). 

On my list of things to do eventually is a "Validate" method for each array type like the C++ library has. That "Validate" method would do the UTF-8 validity check on the buffer so that a consumer can choose *when* they take the performance hit for validating the utf-8.